packages/visx-xychart/src/components/series/BarGroup.tsx

```diff
@@ -0,0 +1,173 @@
+      return { key, data, xAccessor, yAccessor };
+    });
+
+    registerData(dataToRegister);
+    return () => unregisterData(dataKeys);
+  }, [registerData, unregisterData, barSeriesChildren, dataKeys]);
+
+  // create group scale
+  const groupScale = useMemo(
+    () =>
+      scaleBand<string>({
+        domain: sortBars ? [...dataKeys].sort(sortBars) : dataKeys,
+        range: [0, getScaleBandwidth(horizontal ? yScale : xScale)],
+        padding,
+      }),
+    [sortBars, dataKeys, xScale, yScale, horizontal, padding],
+  );
+
+  const { showTooltip, hideTooltip } = useContext(TooltipContext) ?? {};
+  const handleMouseMove = useCallback(
+    (params?: HandlerParams) => {
+      const { svgPoint } = params || {};
+      // invoke showTooltip for each key so all data is available in context,
+      // and let Tooltip find the nearest point among them
+      dataKeys.forEach(key => {
+        const entry = dataRegistry.get(key);
+
+        if (entry && svgPoint && width && height && showTooltip) {
+          const datum = (horizontal ? findNearestDatumY : findNearestDatumX)({
+            point: svgPoint,
+            data: entry.data,
+            xScale,
+            yScale,
+            xAccessor: entry.xAccessor,
+            yAccessor: entry.yAccessor,
+            width,
+            height,
+          });
+          if (datum) {
+            showTooltip({
+              key,
+              ...datum,
+              svgPoint,
+            });
+          }
+        }
+      });
+    },
+    [dataKeys, dataRegistry, horizontal, xScale, yScale, width, height, showTooltip],
+  );
+  useEventEmitter('mousemove', handleMouseMove);
+  useEventEmitter('mouseout', hideTooltip);
+
+  const registryEntries = dataKeys.map(key => dataRegistry.get(key));
+
+  // if scales and data are not available in the registry, bail
+  if (registryEntries.some(entry => entry == null) || !xScale || !yScale || !colorScale) {
+    return null;
+  }
+
+  const [xMin, xMax] = xScale.range().map(Number);
+  const [yMax, yMin] = yScale.range().map(Number);
+
+  // try to figure out the 0 baseline for correct rendering of negative values
+  // we aren't sure if these are numeric scales or not ahead of time
+  const maybeXZero = xScale(0) ?? 0;
+  const maybeYZero = yScale(0) ?? 0;
+  const xZeroPosition = isValidNumber(maybeXZero)
+    ? // if maybeXZero _is_ a number, but the scale is not clamped and it's outside the domain
+      // fallback to the scale's minimum
+      Math.max(maybeXZero, Math.min(xMin, xMax))
+    : Math.min(xMin, xMax);
+  const yZeroPosition = isValidNumber(maybeYZero)
+    ? Math.min(maybeYZero, Math.max(yMin, yMax))
+    : Math.max(yMin, yMax);
+
+  const barThickness = getScaleBandwidth(groupScale);
+
+  return (
+    <g className="visx-bar-group">
+      {registryEntries.map(({ xAccessor, yAccessor, data, key }) =>
+        data.map((datum, index) => {
+          const barLength = horizontal
```

Comment:
similar to suggestion on barstack, these conditions can be moved outside to save `n` times redundant check.